File: src/app/models/employee.ts
import { User } from "./user";

export interface Employee {
  id: number;
  full_name: string;
  gender: string;
  date_of_birth: Date;
  email: string;
  phone_number: string;
  address: string;
  user :User;
  contact_start_date: Date;
  contact_end_date: Date;
  avatar_url: string;
  education: string;
  department: string;
  position: string;
  allowance : number;
  basicSalary  : number;

  // constructor(
  //   // id: number,
  //   name: string,
  //   gender: string,
  //   dateOfBirth: Date,
  //   email: string,
  //   phoneNumber: string,
  //   address: string,
  //   contactStartDate: Date,
  //   contactEndDate: Date,
  //   avatarUrl: string,
  //   education: string,
  //   department: string,
  //   position: string
  // ) {
  //   // this.id = id;
  //   this.name = name;
  //   this.gender = gender;
  //   this.dateOfBirth = dateOfBirth;
  //   this.email = email;
  //   this.phoneNumber = phoneNumber;
  //   this.address = address;
  //   this.contactStartDate = contactStartDate;
  //   this.contactEndDate = contactEndDate;
  //   this.avatarUrl = avatarUrl;
  //   this.education = education;
  //   this.department = department;
  //   this.position = position;
  // }

  
}

File: sys/src/main/java/com/hrms/sys/models/Employee.java
package com.hrms.sys.models;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDate;

@Entity
@Table(name = "employees")
@Data
@Setter
@Getter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Employee {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "full_name")
    private String fullName;

    private String gender;

    @Column(name = "date_of_birth")
    private LocalDate dateOfBirth;

    private String email;

    @Column(name = "phone_number")
    private String phoneNumber;

    @Column(name = "address",length = 300)
    private String address;

    @Column(name = "contact_start_date")
    private LocalDate contactStartDate;

    @Column(name = "contact_end_date")
    private LocalDate contactEndDate;

    @Column(name = "avatar_url")
    private String avatarUrl;


    @OneToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id")
    private User user;

    private String position;

    private String education;

    @ManyToOne
    @JoinColumn(name = "department_id")
    private Department department;

    @Column(name = "remaining_paid_leave_days")
    private float remainingPaidLeaveDays;

    @Column(name = "remaining_remote_days")
    private float remainingRemoteDays;

    @Column(name = "remaining_overtime_hours")
    private float remainingOvertimeHours;

    //Lương cơ bản
    @Column(name = "hourly_wage")
    private float hourlyWage;

    @OneToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "benifit")
    private Benefit benifit;

    @OneToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "payroll")
    private Payroll payroll;



//    private float

}
